codec-redis/src/main/java/io/netty/handler/codec/redis/RedisMessageFactory.java

```diff
@@ -0,0 +1,236 @@
+        Map<ByteBuf, StringRedisMessage> byteBufCache = new HashMap<ByteBuf, StringRedisMessage>();
+        for (String message : messages) {
+            ByteBuf byteBuf = Unpooled.unmodifiableBuffer(
+                    Unpooled.unreleasableBuffer(Unpooled.wrappedBuffer(message.getBytes(CharsetUtil.UTF_8))));
+            StringRedisMessage preloadMessage = new StringRedisMessage(isSimpleString, message);
+            stringsCache.put(message, preloadMessage);
+            byteBufCache.put(byteBuf, preloadMessage);
+        }
+        return new PreloadStringsCache(stringsCache, byteBufCache);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static StringRedisMessage createSimpleString(String content) {
+        ObjectUtil.checkNotNull(content, "content");
+        StringRedisMessage cached = preloadSimpleString.stringCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(true, content);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#SIMPLE_STRING}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createSimpleString(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadSimpleString.byteBufCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(true, content.toString(CharsetUtil.UTF_8));
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} for the given {@code content} using {@link RedisMessageType#ERROR}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createError(String content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadErrors.stringCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(false, content);
+    }
+
+    /**
+     * Creates a {@link StringRedisMessage} from {@code content} using {@link RedisMessageType#ERROR}.
+     *
+     * @param content the message content, must not be {@code null}.
+     * @return a new {@link StringRedisMessage} containing the {@code content}.
+     */
+    public static RedisMessage createError(ByteBuf content) {
+        ObjectUtil.checkNotNull(content, "content");
+        RedisMessage cached = preloadErrors.byteBufCache.get(content);
+        return cached != null ? cached : new StringRedisMessage(false, content.toString(CharsetUtil.UTF_8));
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Ternary operators that can be simplified with || or && [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://garage.netty.io/sonarqube/coding_rules#rule_key=pmd%3ASimplifiedTernary)